// vuestic-ui/dist/web-components/src/components/va-menu/VaMenu.js
import { _ as _sfc_main } from "./VaMenu.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, mergeProps, withCtx, renderSlot, createVNode, withKeys, withModifiers, createSlots } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaMenuList = resolveComponent("VaMenuList");
  const _component_VaDropdownContent = resolveComponent("VaDropdownContent");
  const _component_VaDropdown = resolveComponent("VaDropdown");
  return openBlock(), createBlock(_component_VaDropdown, mergeProps(_ctx.dropdownProps, { ref: "dropdown" }), {
    anchor: withCtx(() => [
      renderSlot(_ctx.$slots, "anchor")
    ]),
    default: withCtx(() => [
      createVNode(_component_VaDropdownContent, { onKeydown: _ctx.onKeydown }, {
        default: withCtx(() => [
          createVNode(_component_VaMenuList, mergeProps({
            onKeydown: _cache[0] || (_cache[0] = withKeys(withModifiers(() => {
            }, ["prevent", "stop"]), ["enter", "space"]))
          }, _ctx.menuListProps, {
            ref: "menuList",
            onSelected: _cache[1] || (_cache[1] = ($event) => {
              _ctx.$emit("selected", $event);
              _ctx.close();
            })
          }), createSlots({ _: 2 }, [
            _ctx.$slots.default ? {
              name: "default",
              fn: withCtx(() => [
                renderSlot(_ctx.$slots, "default")
              ]),
              key: "0"
            } : void 0
          ]), 1040)
        ]),
        _: 3
      }, 8, ["onKeydown"])
    ]),
    _: 3
  }, 16);
}
const _VaMenu = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [...(_sfc_main.components ? Object.values(_sfc_main.components) : []).map((c) => c.styles || []).flat()]]]);
export {
  _VaMenu as _
};
