// vuestic-ui/dist/web-components/src/components/va-textarea/VaTextarea.js
import { _ as _sfc_main } from "./VaTextarea.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createBlock, mergeProps, withCtx, createElementVNode, normalizeClass, withDirectives, vModelText } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ".va-textarea .va-input-wrapper__field {\n  padding-top: 12px;\n  padding-bottom: 12px;\n  align-items: flex-start;\n}\n.va-textarea__resize-wrapper {\n  display: flex;\n  overflow: hidden;\n  width: 100%;\n  align-self: stretch;\n}\n.va-textarea__resize-wrapper--resizable {\n  resize: vertical;\n}\n.va-textarea__textarea {\n  border: 0;\n  flex: 1;\n  font-family: var(--va-font-family);\n  width: 100%;\n  padding: 1px 0;\n  margin: -1px 0;\n  background: transparent;\n  color: currentColor;\n  box-sizing: content-box;\n  min-height: 1.15rem;\n  outline: none;\n  position: relative;\n  resize: none;\n  scrollbar-color: var(--va-secondary) transparent;\n  scrollbar-width: thin;\n  transition: scrollbar-color 0.3s ease-in-out;\n}\n.va-textarea__textarea::-webkit-scrollbar {\n  width: 4px;\n  height: 4px;\n}\n.va-textarea__textarea::-webkit-scrollbar-track {\n  box-shadow: none;\n  border-radius: 10px;\n}\n.va-textarea__textarea::-webkit-scrollbar-thumb {\n  background: var(--va-secondary);\n  opacity: 0.3;\n  border-radius: 2px;\n  -webkit-transition: background 0.3s ease-in-out;\n  transition: background 0.3s ease-in-out;\n}";
const _hoisted_1 = ["rows", "loading", "ariaLabel"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaInputWrapper = resolveComponent("VaInputWrapper");
  return openBlock(), createBlock(_component_VaInputWrapper, mergeProps({ class: "va-textarea" }, _ctx.vaInputWrapperProps, {
    error: _ctx.computedError,
    "error-messages": _ctx.computedErrorMessages
  }), {
    default: withCtx(() => [
      createElementVNode("div", {
        class: normalizeClass(["va-textarea__resize-wrapper", {
          "va-textarea__resize-wrapper--resizable": _ctx.isResizable
        }])
      }, [
        withDirectives(createElementVNode("textarea", mergeProps({
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => _ctx.valueComputed = $event)
        }, { ..._ctx.computedProps, ..._ctx.listeners, ..._ctx.validationAriaAttributes }, {
          style: _ctx.computedStyle,
          rows: _ctx.computedRowsCount,
          loading: _ctx.isLoading,
          ref: "textarea",
          ariaLabel: _ctx.$props.label,
          class: "va-textarea__textarea",
          onFocus: _cache[1] || (_cache[1] = (...args) => _ctx.validationListeners.onFocus && _ctx.validationListeners.onFocus(...args)),
          onBlur: _cache[2] || (_cache[2] = (...args) => _ctx.validationListeners.onBlur && _ctx.validationListeners.onBlur(...args))
        }), null, 16, _hoisted_1), [
          [vModelText, _ctx.valueComputed]
        ])
      ], 2)
    ]),
    _: 1
  }, 16, ["error", "error-messages"]);
}
const _VaTextarea = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [...(_sfc_main.components ? Object.values(_sfc_main.components) : []).map((c) => c.styles || []).flat(), _style_0]]]);
export {
  _VaTextarea as _
};
