// vuestic-ui/dist/web-components/src/components/va-layout/VaLayout.vue_vue_type_script_lang.js
import { defineComponent, computed, watchEffect, useCssVars } from "vue";
import { u as useGridTemplateArea } from "./hooks/useGridTemplateArea.js";
import { u as useLayoutProps, a as useLayout } from "./hooks/useLayout.js";
import { V as VaLayoutArea } from "./components/VaLayoutArea.js";
import { u as useDocument } from "../../composables/useDocument.js";
const areaNames = [
  "top",
  "left",
  "right",
  "bottom"
];
const __default__ = defineComponent({
  name: "VaLayout",
  props: {
    ...useLayoutProps,
    allowBodyScrollOnOverlay: { type: Boolean, default: false }
  },
  emits: [
    "top-overlay-click",
    "left-overlay-click",
    "right-overlay-click",
    "bottom-overlay-click"
  ],
  components: { VaLayoutArea },
  setup(props, { slots }) {
    const { paddings } = useLayout(props);
    const doDisableScroll = computed(() => {
      return !props.allowBodyScrollOnOverlay && areaNames.some((area) => {
        var _a;
        return (_a = props[area]) == null ? void 0 : _a.overlay;
      });
    });
    const document = useDocument();
    watchEffect(() => {
      var _a;
      const overflowParent = (_a = document.value) == null ? void 0 : _a.body;
      if (!overflowParent) {
        return;
      }
      if (doDisableScroll.value) {
        overflowParent.style.overflow = "hidden";
      } else {
        overflowParent.style.overflow = "";
      }
    });
    return {
      areaNames,
      templateArea: useGridTemplateArea(props),
      verticalTemplate: computed(() => {
        return [
          slots.top ? "min-content" : "0fr",
          "1fr",
          slots.bottom ? "min-content" : "0fr"
        ].filter(Boolean).join(" ");
      }),
      horizontalTemplate: computed(() => {
        return [
          slots.left ? "min-content" : "0fr",
          "1fr",
          slots.right ? "min-content" : "0fr"
        ].filter(Boolean).join(" ");
      }),
      paddings
    };
  }
});
const __injectCSSVars__ = () => {
  useCssVars((_ctx) => ({
    "03463818": _ctx.horizontalTemplate,
    "948cba74": _ctx.verticalTemplate,
    "42c7efdd": _ctx.templateArea,
    "1bd1c7e4": _ctx.paddings.top + "px",
    "8c726468": _ctx.paddings.bottom + "px",
    "365ba2f0": _ctx.paddings.left + "px",
    "0bad9032": _ctx.paddings.right + "px"
  }));
};
const __setup__ = __default__.setup;
__default__.setup = __setup__ ? (props, ctx) => {
  __injectCSSVars__();
  return __setup__(props, ctx);
} : __injectCSSVars__;
export {
  __default__ as _
};
