// vuestic-ui/dist/es/src/components/va-alert/VaAlert.vue_vue_type_script_lang.js
import { defineComponent, computed } from "vue";
import { u as useAlertStyles } from "./useAlertStyles.js";
import { V as VaIcon } from "../va-icon/index.js";
import { u as useStatefulEmits, a as useStatefulProps, b as useStateful } from "../../composables/useStateful.js";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.js";
import { u as useTranslation } from "../../composables/useTranslation.js";
const _sfc_main = defineComponent({
  name: "VaAlert",
  components: { VaIcon },
  emits: [...useStatefulEmits],
  props: {
    ...useStatefulProps,
    ...useComponentPresetProp,
    modelValue: { type: Boolean, default: true },
    stateful: { type: Boolean, default: true },
    color: { type: String, default: "primary" },
    textColor: { type: String, default: "" },
    title: { type: String, default: "" },
    description: { type: String, default: "" },
    icon: { type: String, default: "" },
    closeText: { type: String, default: "" },
    closeIcon: { type: String, default: "close" },
    closeable: { type: Boolean, default: false },
    dense: { type: Boolean, default: false },
    outline: { type: Boolean, default: false },
    center: { type: Boolean, default: false },
    borderColor: { type: String, default: "" },
    border: {
      type: String,
      default: "",
      validator: (value) => ["top", "right", "bottom", "left", ""].includes(value)
    }
  },
  setup(props, { slots, emit }) {
    const {
      contentStyle,
      titleStyle,
      alertStyle,
      borderStyle
    } = useAlertStyles(props);
    const { valueComputed } = useStateful(props, emit);
    const hide = () => {
      valueComputed.value = false;
    };
    const show = () => {
      valueComputed.value = true;
    };
    const hasIcon = computed(() => props.icon || slots.icon);
    const hasTitle = computed(() => props.title || slots.title);
    const borderClass = computed(() => `va-alert__border--${props.border}`);
    const { tp, t } = useTranslation();
    return {
      tp,
      t,
      contentStyle,
      titleStyle,
      alertStyle,
      borderStyle,
      valueComputed,
      hasIcon,
      hasTitle,
      borderClass,
      hide,
      show
    };
  }
});
export {
  _sfc_main as _
};

import '../../../VaAlert.css';
