// vuestic-ui/dist/es/src/components/va-layout/VaLayout.vue_vue_type_script_lang.js
import { defineComponent, computed, watchEffect } from "vue";
import { u as useGridTemplateArea } from "./hooks/useGridTemplateArea.js";
import { u as useLayoutProps, a as useLayout } from "./hooks/useLayout.js";
import { V as VaLayoutArea } from "./components/VaLayoutArea.js";
import { u as useDocument } from "../../composables/useDocument.js";
const areaNames = [
  "top",
  "left",
  "right",
  "bottom"
];
const _sfc_main = defineComponent({
  name: "VaLayout",
  props: {
    ...useLayoutProps,
    allowBodyScrollOnOverlay: { type: Boolean, default: false }
  },
  emits: [
    "top-overlay-click",
    "left-overlay-click",
    "right-overlay-click",
    "bottom-overlay-click"
  ],
  components: { VaLayoutArea },
  setup(props, { slots }) {
    const { paddings } = useLayout(props);
    const doDisableScroll = computed(() => {
      return !props.allowBodyScrollOnOverlay && areaNames.some((area) => {
        var _a;
        return (_a = props[area]) == null ? void 0 : _a.overlay;
      });
    });
    const document = useDocument();
    watchEffect(() => {
      var _a;
      const overflowParent = (_a = document.value) == null ? void 0 : _a.body;
      if (!overflowParent) {
        return;
      }
      if (doDisableScroll.value) {
        overflowParent.style.overflow = "hidden";
      } else {
        overflowParent.style.overflow = "";
      }
    });
    return {
      areaNames,
      templateArea: useGridTemplateArea(props),
      verticalTemplate: computed(() => {
        return [
          slots.top ? "min-content" : "0fr",
          "1fr",
          slots.bottom ? "min-content" : "0fr"
        ].filter(Boolean).join(" ");
      }),
      horizontalTemplate: computed(() => {
        return [
          slots.left ? "min-content" : "0fr",
          "1fr",
          slots.right ? "min-content" : "0fr"
        ].filter(Boolean).join(" ");
      }),
      paddings
    };
  }
});
export {
  _sfc_main as _
};

import '../../../VaLayout.css';
