// vuestic-ui/dist/web-components/src/components/va-collapse/VaCollapse.js
import { _ as _sfc_main } from "./VaCollapse.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, withKeys, renderSlot, normalizeProps, guardReactiveProps, mergeProps, createBlock, createCommentVNode, toDisplayString, createVNode, normalizeStyle } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
const _style_0 = ":root,\n:host {\n  --va-collapse-padding: 0.75rem;\n  --va-collapse-gap: 0.5rem;\n  --va-collapse-transition: all 0.3s linear;\n  --va-collapse-popout-margin: 0.5rem -0.5rem;\n  --va-collapse-inset-margin: 0.5rem 0.5rem;\n  --va-collapse-body-transition: opacity 0.1s linear;\n  --va-collapse-body-wrapper-transition: height 0.3s linear, background 0.3s linear;\n  --va-collapse-body-width: 100%;\n  --va-collapse-header-content-cursor: pointer;\n  --va-collapse-header-content-background-color: #f5f8f9;\n  --va-collapse-header-content-box-shadow: var(--va-box-shadow);\n  --va-collapse-header-content-border-radius: 0.375rem;\n  --va-collapse-header-content-padding-top: 0.75rem;\n  --va-collapse-header-content-padding-bottom: 0.75rem;\n  --va-collapse-header-content-padding-left: 1rem;\n  --va-collapse-header-content-text-font-weight: 600;\n  --va-collapse-header-content-icon-min-width: 1.5rem;\n  --va-collapse-header-content-icon-margin-left: 0.5rem;\n  --va-collapse-header-content-icon-margin-right: 0.5rem;\n  --va-collapse-header-content-icon-color: var(--va-background-element);\n  --va-collapse-expand-icon-transition: transform 0.3s linear;\n  /* Solid */\n  --va-collapse-solid-box-shadow: var(--va-box-shadow);\n  --va-collapse-solid-border-radius: 0.375rem;\n  --va-collapse-solid-header-content-background-color: #f5f8f9;\n  --va-collapse-solid-header-content-box-shadow: none;\n  --va-collapse-solid-header-content-border-radius: 0.375rem;\n  --va-collapse-solid-header-content-transition: background-color ease-in 0.3s;\n  --va-collapse-solid-body-border-radius: 0 0 0.375rem 0.375rem;\n  --va-collapse-solid-body-margin-top: 0;\n}\n.va-collapse {\n  transition: var(--va-collapse-transition, var(--va-swing-transition));\n  font-family: var(--va-font-family);\n  display: flex;\n  flex-direction: column;\n}\n.va-collapse__body-wrapper {\n  transition: var(--va-collapse-body-wrapper-transition);\n  overflow: auto;\n}\n.va-collapse__body-wrapper--bordered {\n  border-bottom: 1px solid var(--va-background-border);\n  box-sizing: content-box;\n}\n.va-collapse--colored-header:not(.va-collapse--expanded) .va-collapse__body-wrapper--bordered {\n  border-bottom: none;\n}\n.va-collapse--colored-body.va-collapse--expanded .va-collapse__body-wrapper--bordered {\n  border-bottom: none;\n}\n.va-collapse__body {\n  top: 0;\n  left: 0;\n  width: var(--va-collapse-body-width);\n  transition: var(--va-collapse-body-transition);\n  opacity: 0;\n}\n.va-collapse__content {\n  padding: var(--va-collapse-padding);\n  padding-top: calc(var(--va-collapse-padding) / 1.5);\n  box-sizing: border-box;\n}\n.va-collapse__content:empty {\n  padding: 0;\n}\n.va-collapse__header {\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  gap: var(--va-collapse-gap);\n  cursor: var(--va-collapse-header-content-cursor);\n  padding: var(--va-collapse-padding);\n  transition: all 0.2s ease-in-out;\n}\n.va-collapse__header__text {\n  width: 100%;\n  font-weight: var(--va-collapse-header-content-text-font-weight);\n}\n.va-collapse__header:focus-visible {\n  outline: 2px solid var(--va-focus) !important;\n  border-radius: var(--va-collapse-header-content-border-radius);\n}\n.va-collapse--expanded .va-collapse__body {\n  opacity: 1;\n}\n.va-collapse__expand-icon {\n  transition: var(--va-collapse-expand-icon-transition);\n}\n.va-collapse__expand-icon--expanded {\n  transform: rotate(180deg);\n}\n.va-collapse--popout {\n  margin: var(--va-collapse-popout-margin);\n}\n.va-collapse--inset {\n  margin: var(--va-collapse-inset-margin);\n}\n.va-collapse--disabled {\n  cursor: default;\n  opacity: 0.4;\n  pointer-events: none;\n  -webkit-user-select: none;\n     -moz-user-select: none;\n          user-select: none;\n}\n.va-collapse--height-changing .va-collapse__body-wrapper {\n  overflow: hidden;\n}";
const _hoisted_1 = { class: "va-collapse__header__text" };
const _hoisted_2 = ["id", "aria-labelledby"];
const _hoisted_3 = { class: "va-collapse__content" };
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_icon = resolveComponent("va-icon");
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-collapse", _ctx.computedClasses])
  }, [
    createElementVNode("div", {
      class: "va-collapse__header-wrapper",
      role: "heading",
      onClick: _cache[0] || (_cache[0] = (...args) => _ctx.toggle && _ctx.toggle(...args)),
      onKeydown: [
        _cache[1] || (_cache[1] = withKeys((...args) => _ctx.toggle && _ctx.toggle(...args), ["enter"])),
        _cache[2] || (_cache[2] = withKeys((...args) => _ctx.toggle && _ctx.toggle(...args), ["space"]))
      ]
    }, [
      renderSlot(_ctx.$slots, "header", normalizeProps(guardReactiveProps({
        value: _ctx.computedModelValue,
        bind: _ctx.headerAttributes,
        attributes: _ctx.headerAttributes,
        attrs: _ctx.headerAttributes,
        iconAttrs: {
          class: [
            "va-collapse__expand-icon",
            _ctx.computedModelValue ? "a-collapse__expand-icon--expanded" : "a-collapse__expand-icon--collapsed"
          ]
        },
        text: _ctx.header
      })), () => [
        createElementVNode("div", mergeProps(_ctx.headerAttributes, {
          class: "va-collapse__header",
          style: _ctx.headerStyle
        }), [
          _ctx.icon ? (openBlock(), createBlock(_component_va_icon, {
            key: 0,
            class: "va-collapse__header__icon",
            name: _ctx.icon
          }, null, 8, ["name"])) : createCommentVNode("", true),
          renderSlot(_ctx.$slots, "header-content", normalizeProps(guardReactiveProps({ header: _ctx.header })), () => [
            createElementVNode("div", _hoisted_1, toDisplayString(_ctx.header), 1)
          ]),
          renderSlot(_ctx.$slots, "expand-icon", {}, () => [
            createVNode(_component_va_icon, {
              class: normalizeClass(["va-collapse__expand-icon", _ctx.computedModelValue ? "va-collapse__expand-icon--expanded" : "va-collapse__expand-icon--collapsed"]),
              name: "va-arrow-down"
            }, null, 8, ["class"])
          ])
        ], 16)
      ])
    ], 32),
    createElementVNode("div", {
      class: normalizeClass(["va-collapse__body-wrapper", {
        "va-collapse__body-wrapper--bordered": !_ctx.$slots.body && !_ctx.$slots.header
      }]),
      style: normalizeStyle(_ctx.contentStyle),
      onTransitionend: _cache[3] || (_cache[3] = (...args) => _ctx.onTransitionEnd && _ctx.onTransitionEnd(...args))
    }, [
      createElementVNode("div", {
        class: "va-collapse__body",
        ref: "body",
        role: "region",
        id: _ctx.panelIdComputed,
        "aria-labelledby": _ctx.headerIdComputed
      }, [
        renderSlot(_ctx.$slots, "body", {}, () => [
          createElementVNode("div", _hoisted_3, [
            renderSlot(_ctx.$slots, "default", {}, () => [
              renderSlot(_ctx.$slots, "content")
            ])
          ])
        ])
      ], 8, _hoisted_2)
    ], 38)
  ], 2);
}
const _VaCollapse = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["styles", [...(_sfc_main.components ? Object.values(_sfc_main.components) : []).map((c) => c.styles || []).flat(), _style_0]]]);
export {
  _VaCollapse as _
};
