// vuestic-ui/dist/esm-node/src/components/va-modal/VaModal.mjs
import { w as withConfigTransport } from "../../services/config-transport/withConfigTransport.mjs";
import { _ as _sfc_main } from "./VaModal.vue_vue_type_script_lang.mjs";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, mergeProps, renderSlot, normalizeProps, guardReactiveProps, createCommentVNode, createBlock, Teleport, createVNode, withCtx, createElementVNode, withKeys, Fragment, toDisplayString, createTextVNode } from "vue";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.mjs";
const VaModal_vue_vue_type_style_index_0_lang = "";
const _hoisted_1 = ["aria-labelledby"];
const _hoisted_2 = { class: "va-modal__container" };
const _hoisted_3 = { key: 0 };
const _hoisted_4 = {
  key: 1,
  class: "va-modal__header"
};
const _hoisted_5 = {
  key: 2,
  class: "va-modal__message"
};
const _hoisted_6 = {
  key: 3,
  class: "va-modal__message"
};
const _hoisted_7 = {
  key: 4,
  class: "va-modal__footer"
};
const _hoisted_8 = {
  key: 5,
  class: "va-modal__footer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_icon = resolveComponent("va-icon");
  const _component_va_button = resolveComponent("va-button");
  const _component_WithTransition = resolveComponent("WithTransition");
  return openBlock(), createElementBlock("div", {
    ref: "rootElement",
    class: normalizeClass(["va-modal-entry", _ctx.$props.anchorClass]),
    role: "dialog",
    "aria-modal": "true",
    "aria-labelledby": _ctx.title,
    style: normalizeStyle(`--va-z-index-computed: ${String(_ctx.zIndexComputed)}`)
  }, [
    _ctx.$slots.anchor ? (openBlock(), createElementBlock("div", mergeProps({
      key: 0,
      class: "va-modal__anchor"
    }, _ctx.teleportFromAttrs), [
      renderSlot(_ctx.$slots, "anchor", normalizeProps(guardReactiveProps(_ctx.slotBind)))
    ], 16)) : createCommentVNode("", true),
    (openBlock(), createBlock(Teleport, {
      to: _ctx.attachElement,
      disabled: _ctx.$props.disableAttachment
    }, [
      createVNode(_component_WithTransition, mergeProps({
        name: "va-modal",
        isTransition: !_ctx.$props.withoutTransitions,
        appear: "",
        duration: 300
      }, { ..._ctx.$attrs, ..._ctx.teleportedAttrs }, {
        onBeforeEnter: _ctx.onBeforeEnterTransition,
        onAfterEnter: _ctx.onAfterEnterTransition,
        onBeforeLeave: _ctx.onBeforeLeaveTransition,
        onAfterLeave: _ctx.onAfterLeaveTransition
      }), {
        default: withCtx(() => [
          _ctx.valueComputed ? (openBlock(), createElementBlock("div", {
            key: 0,
            class: normalizeClass(["va-modal", _ctx.computedClass])
          }, [
            _ctx.$props.overlay ? (openBlock(), createElementBlock("div", {
              key: 0,
              class: normalizeClass(["va-modal__overlay", _ctx.computedOverlayClass]),
              style: normalizeStyle(_ctx.computedOverlayStyles)
            }, null, 6)) : createCommentVNode("", true),
            createElementVNode("div", _hoisted_2, [
              createElementVNode("div", {
                ref: "modalDialog",
                class: "va-modal__dialog",
                style: normalizeStyle(_ctx.computedDialogStyle)
              }, [
                _ctx.$props.fullscreen || _ctx.$props.closeButton ? (openBlock(), createBlock(_component_va_icon, {
                  key: 0,
                  name: "va-close",
                  class: normalizeClass(["va-modal__close", { "va-modal__close--fullscreen": _ctx.$props.fullscreen }]),
                  role: "button",
                  "aria-label": _ctx.tp(_ctx.$props.ariaCloseLabel),
                  tabindex: "0",
                  onClick: _ctx.cancel,
                  onKeydown: [
                    withKeys(_ctx.cancel, ["space"]),
                    withKeys(_ctx.cancel, ["enter"])
                  ]
                }, null, 8, ["class", "aria-label", "onClick", "onKeydown"])) : createCommentVNode("", true),
                createElementVNode("div", {
                  class: "va-modal__inner",
                  style: normalizeStyle({ maxWidth: _ctx.$props.maxWidth, maxHeight: _ctx.$props.maxHeight })
                }, [
                  _ctx.$slots.content ? (openBlock(), createElementBlock("div", _hoisted_3, [
                    renderSlot(_ctx.$slots, "content", normalizeProps(guardReactiveProps(_ctx.slotBind)))
                  ])) : createCommentVNode("", true),
                  !_ctx.$slots.content ? (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                    _ctx.title ? (openBlock(), createElementBlock("div", {
                      key: 0,
                      class: "va-modal__title",
                      style: normalizeStyle({ color: _ctx.getColor("primary") })
                    }, toDisplayString(_ctx.$props.title), 5)) : createCommentVNode("", true),
                    _ctx.$slots.header ? (openBlock(), createElementBlock("div", _hoisted_4, [
                      renderSlot(_ctx.$slots, "header", normalizeProps(guardReactiveProps(_ctx.slotBind)))
                    ])) : createCommentVNode("", true),
                    _ctx.$props.message ? (openBlock(), createElementBlock("div", _hoisted_5, toDisplayString(_ctx.$props.message), 1)) : createCommentVNode("", true),
                    _ctx.$slots.default ? (openBlock(), createElementBlock("div", _hoisted_6, [
                      renderSlot(_ctx.$slots, "default", normalizeProps(guardReactiveProps(_ctx.slotBind)))
                    ])) : createCommentVNode("", true),
                    (_ctx.$props.cancelText || _ctx.$props.okText) && !_ctx.$props.hideDefaultActions ? (openBlock(), createElementBlock("div", _hoisted_7, [
                      _ctx.$props.cancelText ? (openBlock(), createBlock(_component_va_button, {
                        key: 0,
                        preset: "secondary",
                        color: "secondary",
                        class: "va-modal__default-cancel-button",
                        onClick: _ctx.cancel
                      }, {
                        default: withCtx(() => [
                          createTextVNode(toDisplayString(_ctx.tp(_ctx.$props.cancelText)), 1)
                        ]),
                        _: 1
                      }, 8, ["onClick"])) : createCommentVNode("", true),
                      createVNode(_component_va_button, { onClick: _ctx.ok }, {
                        default: withCtx(() => [
                          createTextVNode(toDisplayString(_ctx.tp(_ctx.$props.okText)), 1)
                        ]),
                        _: 1
                      }, 8, ["onClick"])
                    ])) : createCommentVNode("", true),
                    _ctx.$slots.footer ? (openBlock(), createElementBlock("div", _hoisted_8, [
                      renderSlot(_ctx.$slots, "footer", normalizeProps(guardReactiveProps(_ctx.slotBind)))
                    ])) : createCommentVNode("", true)
                  ], 64)) : createCommentVNode("", true)
                ], 4)
              ], 4)
            ])
          ], 2)) : createCommentVNode("", true)
        ]),
        _: 3
      }, 16, ["isTransition", "onBeforeEnter", "onAfterEnter", "onBeforeLeave", "onAfterLeave"])
    ], 8, ["to", "disabled"]))
  ], 14, _hoisted_1);
}
const _VaModal = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
const VaModal = withConfigTransport(_VaModal);
export {
  VaModal as V
};
