// vuestic-ui/dist/esm-node/src/components/va-menu/VaMenu.vue_vue_type_script_lang.mjs
import { defineComponent, ref, nextTick } from "vue";
import { V as VaDropdown } from "../va-dropdown/index.mjs";
import { V as VaMenuList } from "../va-menu-list/index.mjs";
import { a as focusFirstFocusableChild } from "../../utils/focus.mjs";
import { u as unwrapEl } from "../../utils/unwrapEl.mjs";
import { V as VaDropdownContent } from "../va-dropdown/components/VaDropdownContent/index.mjs";
import { u as useImmediateFocus } from "../../composables/useImmediateFocus.mjs";
import { e as extractComponentProps, a as extractComponentEmits } from "../../utils/component-options/extract-component-options.mjs";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";
import { f as filterComponentProps } from "../../utils/component-options/filter-props.mjs";
const VaMenuListProps = extractComponentProps(VaMenuList);
const VaMenuListEmits = extractComponentEmits(VaMenuList);
const VaDropdownProps = extractComponentProps(VaDropdown);
const VaDropdownEmits = extractComponentEmits(VaDropdown);
const _sfc_main = defineComponent({
  name: "VaMenu",
  components: { VaDropdown, VaDropdownContent, VaMenuList },
  props: {
    ...useComponentPresetProp,
    ...VaMenuListProps,
    ...VaDropdownProps,
    stickToEdges: { type: Boolean, default: true }
  },
  emits: [
    ...VaDropdownEmits,
    ...VaMenuListEmits
  ],
  setup() {
    const menuList = ref();
    const dropdown = ref();
    useImmediateFocus(menuList);
    const close = () => {
      var _a;
      (_a = dropdown.value) == null ? void 0 : _a.hide();
      nextTick(() => {
        var _a2;
        const el = unwrapEl((_a2 = dropdown.value) == null ? void 0 : _a2.anchorRef);
        if (el) {
          focusFirstFocusableChild(el);
        }
      });
    };
    const onKeydown = (event) => {
      if (event.key === "Escape") {
        close();
      }
      if (event.key === "ArrowDown" || event.key === "ArrowUp") {
        event.preventDefault();
      }
    };
    return {
      onKeydown,
      dropdown,
      menuList,
      menuListProps: filterComponentProps(VaMenuListProps),
      dropdownProps: filterComponentProps(VaDropdownProps),
      close
    };
  }
});
export {
  _sfc_main as _
};
