// vuestic-ui/dist/web-components/src/components/va-layout/components/VaLayoutArea.vue_vue_type_script_lang.js
import { defineComponent, computed, useCssVars } from "vue";
import { V as VaLayoutAbsoluteWrapper } from "./VaLayoutAbsoluteWrapper.js";
import { V as VaLayoutFixedWrapper } from "./VaLayoutFixedWrapper.js";
const __default__ = defineComponent({
  name: "VaLayoutArea",
  components: {
    VaLayoutAbsoluteWrapper,
    VaLayoutFixedWrapper
  },
  props: {
    area: { type: String, required: true },
    config: { type: Object, required: true }
  },
  emits: ["overlay-click"],
  setup(props) {
    return {
      absolute: computed(() => props.config.absolute || false),
      fixed: computed(() => props.config.fixed || false),
      overlay: computed(() => props.config.overlay || false),
      // Content z-index is always 0, other areas must have bigger z-index by 1
      zIndex: computed(() => (props.config.order || 0) + 1)
    };
  }
});
const __injectCSSVars__ = () => {
  useCssVars((_ctx) => ({
    "2f768c34": _ctx.$props.area,
    "2581d5a1": _ctx.zIndex,
    "3f7b55ff": _ctx.zIndex - 1
  }));
};
const __setup__ = __default__.setup;
__default__.setup = __setup__ ? (props, ctx) => {
  __injectCSSVars__();
  return __setup__(props, ctx);
} : __injectCSSVars__;
export {
  __default__ as _
};
