// vuestic-ui/dist/web-components/src/components/va-layout/components/VaLayoutFixedWrapper.js
import { _ as __default__ } from "./VaLayoutFixedWrapper.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeStyle, createVNode, normalizeClass, withCtx, renderSlot } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const _style_0 = ".va-layout-fixed-wrapper {\n  position: relative;\n  flex: 1;\n}\n.va-layout-fixed-wrapper__content {\n  position: fixed;\n  width: var(--121be6a3);\n  height: var(--1791f3ea);\n}\n@media print {\n.va-layout-fixed-wrapper__content {\n    position: relative !important;\n    height: -moz-max-content !important;\n    height: max-content !important;\n    width: -moz-max-content !important;\n    width: max-content !important;\n}\n}\n.va-layout-fixed-wrapper__content--top {\n  top: 0;\n}\n.va-layout-fixed-wrapper__content--bottom {\n  bottom: 0;\n}\n.va-layout-fixed-wrapper__content--right {\n  right: 0;\n}\n.va-layout-fixed-wrapper__content--left {\n  left: 0;\n}";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaResizeObserver = resolveComponent("VaResizeObserver");
  return openBlock(), createElementBlock("div", {
    class: "va-layout-fixed-wrapper",
    style: normalizeStyle([{
      height: _ctx.size && _ctx.direction === "vertical" ? _ctx.size.height + "px" : "auto",
      width: _ctx.size && _ctx.direction === "horizontal" ? _ctx.size.width + "px" : "auto"
    }])
  }, [
    createVNode(_component_VaResizeObserver, {
      class: normalizeClass(["va-layout-fixed-wrapper__content", `va-layout-fixed-wrapper__content--${_ctx.area}`]),
      onResize: _cache[0] || (_cache[0] = ($event) => _ctx.size = $event)
    }, {
      default: withCtx(() => [
        renderSlot(_ctx.$slots, "default")
      ]),
      _: 3
    }, 8, ["class"])
  ], 4);
}
const VaLayoutFixedWrapper = /* @__PURE__ */ _export_sfc(__default__, [["render", _sfc_render], ["styles", [...(__default__.components ? Object.values(__default__.components) : []).map((c) => c.styles || []).flat(), _style_0]]]);
export {
  VaLayoutFixedWrapper as V
};
