// vuestic-ui/dist/es/src/components/va-layout/components/VaLayoutFixedWrapper.js
import { _ as _sfc_main } from "./VaLayoutFixedWrapper.vue_vue_type_script_lang.js";
import { resolveComponent, openBlock, createElementBlock, normalizeStyle, createVNode, normalizeClass, withCtx, renderSlot } from "vue";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.js";
const VaLayoutFixedWrapper_vue_vue_type_style_index_0_lang = "";
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaResizeObserver = resolveComponent("VaResizeObserver");
  return openBlock(), createElementBlock("div", {
    class: "va-layout-fixed-wrapper",
    style: normalizeStyle([[{
      height: _ctx.size && _ctx.direction === "vertical" ? _ctx.size.height + "px" : "auto",
      width: _ctx.size && _ctx.direction === "horizontal" ? _ctx.size.width + "px" : "auto"
    }], `--va-styles-width: ${String(_ctx.styles.width)};--va-styles-height: ${String(_ctx.styles.height)}`])
  }, [
    createVNode(_component_VaResizeObserver, {
      class: normalizeClass(["va-layout-fixed-wrapper__content", `va-layout-fixed-wrapper__content--${_ctx.area}`]),
      onResize: _cache[0] || (_cache[0] = ($event) => _ctx.size = $event)
    }, {
      default: withCtx(() => [
        renderSlot(_ctx.$slots, "default")
      ]),
      _: 3
    }, 8, ["class"])
  ], 4);
}
const VaLayoutFixedWrapper = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  VaLayoutFixedWrapper as V
};

import '../../../../VaLayoutFixedWrapper.css';
